crates/agent/src/publications/builds.rs

```diff
@@ -21,6 +21,7 @@ pub struct BuildOutput {
 #[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
 pub struct IncompatibleCollection {
     pub collection: String,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
```

Comment:
There's no good way to populate `affected_materializations` in this scenario because we haven't yet expanded the catalog when we call `validate_transition`. So I'm thinking that "affected_" is not a great name here, and perhaps we can consider renaming it to `rejected_by_materializations`. Nothing actually reads this value, though, so I'm thinking we deal with renaming it in a subsequent PR so as not to delay this one.